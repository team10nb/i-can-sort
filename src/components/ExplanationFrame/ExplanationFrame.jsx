/*
    Author: Yijie Lu
*/
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import HelpOutlineOutlinedIcon from '@material-ui/icons/HelpOutlineOutlined';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import HighlightOffIcon from '@material-ui/icons/HighlightOff';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import RotateLeftIcon from '@material-ui/icons/RotateLeft';
import Backdrop from '@material-ui/core/Backdrop';


const useStyles = makeStyles((theme) => ({
    backdrop: {
      zIndex: theme.zIndex.drawer + 1,
      color: "#212121",
      marginLeft:"-500px",
      marginTop:"-155px",
      height:"1000px",
      width:"2000px",
    },

    card: {
      maxWidth: 580,
      height: 570,
      borderRadius:"30px",
    },

    // avatar: {
    //   "& > *": {
    //     marginTop: theme.spacing(1),
    //     marginRight: theme.spacing(100),
    //   },
    // },
    icon:{
      color: "#696969"
    },
    actionArea:{
      minWidth: 395,
      minHeight: 570,
    },

    button:{
      display: "flex",
        "& > *": {
            marginBottom: theme.spacing(2.5),
            marginLeft: theme.spacing(3),
        },
        "& > * + *": {
            marginLeft: theme.spacing(55),
        },
     },

     resetbutton:{    
        "& > *": {
           marginRight: theme.spacing(61),
            marginTop: theme.spacing(0),
        },
     },
  }));
  
  

  

  function getStepContent(step,props) {
    const {color, logo0, logo1, logo2, logo3, logo4, logoFianl, stepNumber, content0, content1,
            content2, content3, content4, content5} = props;
    if (stepNumber === 5){
        switch (step) {
            case 0:
              return (
                <CardActionArea style={{ width: "580px", height: "415px", marginBottom:"15px" }}>
                    <img src={logo0} alt="logo0" width="587" heigh="300" />
                </CardActionArea>
      
              );
            case 1:
              return (
                <CardActionArea style={{ width: "580px", height: "415px", marginBottom:"15px" }}>
                    <img src={logo1} alt="logo1" width="587" heigh="300" />
                </CardActionArea>
              );
            case 2:
            return (
                <CardActionArea style={{ width: "580px", height: "415px", marginBottom:"15px" }}>
                <img src={logo2} alt="logo2" width="587" heigh="300" />
                    <CardContent>
                      {content2}                         
                    </CardContent>
                </CardActionArea>
            );
            case 3:
              return (
                <CardActionArea style={{ width: "580px", height: "415px", marginBottom:"15px" }}>
                    <img src={logo3} alt="logo3" width="587" heigh="300" />
                </CardActionArea>
              );
            default:
              return 'Unknown step';
          }

    }else if (stepNumber === 3){
        switch (step) {
            case 0:
              return (
                <CardActionArea  style={{ height:'415px', marginBottom:"15px"}}>
                  <img src={logo0} alt="logo0" width='587px'  />
                  <CardContent >              
                      {content0}                           
                  </CardContent>
                </CardActionArea>
      
              );
            case 1:
              return (
                <CardActionArea style={{ width: "580px", height: "415px", marginBottom:"15px" }}>
                  <img src={logo1} alt="logo1"  width='587px' />
                  <CardContent >   
                      {content1}                              
                  </CardContent>
                </CardActionArea>
              );
            default:
              return 'Unknown step';
          }
    }else if (stepNumber === 2){
      switch (step) {
        case 0:
          return (
            <CardActionArea  style={{ height:'415px', marginBottom:"15px"}}>
              <img src={logo0} alt="logo0" width='587px'  />
            </CardActionArea>
  
          );
        default:
          return 'Unknown step';
      }

    }
    
  }


 
  export  default function ExplanationFrame(props) {
    const {color, logo0, logo1, logoFinal, stepNumber,content0,content1,contentFinal,picHeight,contentMarginBottom,contentMarginTop,avatarMarginLeft,avatarMarginTop} = props;

    const classes = useStyles();

    const [open, setOpen] = React.useState(false);
 
    const handleClickOpen = () => {
      setOpen(true);
    };

    const handleClose = () => {
      setOpen(false);
    };

    const [activeStep, setActiveStep] = React.useState(0);  

    const steps = stepNumber;
    
    const handleNext = () => {
    setActiveStep((prevActiveStep) => prevActiveStep + 1);     
    };

    const handleBack = () => {
      setActiveStep((prevActiveStep) => prevActiveStep - 1);
    };
    const handleReset = () => {
      setActiveStep(0);
    };

    const card = (
      // <Dialog onClose={handleClose} aria-labelledby="simple-dialog-title" open={open} classes = {classes.dialog}>
      //   <Fade in={open}>
      <Backdrop className={classes.backdrop} open={open} >
          <Card className={classes.card}>          
            <div style={{marginLeft:avatarMarginLeft, marginTop:avatarMarginTop}}>
                <IconButton aria-label="close"  onClick={handleClose} >
                  <HighlightOffIcon  style={{ color: color, opacity: 0.8, fontSize: 40  } } />
                </IconButton>
            </div> 
              {activeStep === steps - 1 ? (
                <div>
                  <div>
                <CardActionArea  style={{ height:picHeight, marginBottom:"15px"}}>
                  <img src={logoFinal} alt="logoFinal"  width= '587px' />
                  <CardContent style={{marginBottom:contentMarginBottom, marginTop:contentMarginTop}}>
                    <Typography variant="body2" color="textSecondary" component="p">
                        {contentFinal} 
                    </Typography>             
                  </CardContent>
                </CardActionArea>    
              </div>
              <div className = {classes.button}>
                 <IconButton  onClick={handleReset} style={{ backgroundColor: color, opacity:0.5, }} size='small'><RotateLeftIcon style={{ color: 'white', fontSize:'40px'}}/></IconButton>          
               </div>
                </div>              
              ) : (
                <div>
                  <div>
                    {getStepContent(activeStep,props)}                    
                  </div>
                  <div className = {classes.button}>               
                   <IconButton  disabled={activeStep === 0} onClick={handleBack}  style={{ backgroundColor: color, opacity:0.5, }} size='small'><ChevronLeftIcon style={{ color: 'white', fontSize:'40px'}}/></IconButton>           
                  <IconButton  onClick={handleNext} style={{ backgroundColor: color, opacity:0.5, }} size='small'><ChevronRightIcon style={{ color: 'white', fontSize:'40px'}}/></IconButton>             
                 </div>
                </div>                
                )}
          </Card> 
      </Backdrop> 
    );

  
    return(
      <div style={{height:"48px"}}>
        <Tooltip title="tutorial" placement="bottom" arrow>
          <IconButton aria-label="help"  onClick={handleClickOpen}>
            <HelpOutlineOutlinedIcon className = {classes.icon}/>   
          </IconButton>
        </Tooltip>
        {card}
      </div>
    )
  }

