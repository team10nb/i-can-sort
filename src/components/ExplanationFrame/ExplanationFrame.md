

### **Developer**

Yijie Lu, Shiliang Chen

Reference: MATERIAL-UI (https://material-ui.com)



###  **Introduction**

It is a interface for all the explanation window. By passing the props, using it as a interface.

###  **Import**

```html
import { makeStyles } from '@material-ui/core/styles';
import logoFinal from "../../../../resource/ForLoop.png";
```

###  **Props**

| Name | Type | Default | Description |
| ---- | ---- | ------- | ----------- |
|color              |  string      |   \ |     the color of the explanation window            |
|logon              |  string      |   \ |     pictures used in the explanation window        |
|contentn           |  JSX.Element |   \ |   contents uesd in the explanation window          |
|stepNumber         |  number      |   \ | the number of pages used in the explanation window |
|picHeight          |  string      |   \ | the height a picture defined                       |
|contentMarginBottom| stirng       |  \  | the marginbottom a content defined                 |
|contentMarginTop   | stirng       |  \  | the margintop a content defined                    |
|avatarMarginLeft   | stirng       |  \  | the marginleft a avatar defined                    |
|avatarMarginTop    | stirng       |  \  | the margintop a avatar defined                     |


###  **Methods**

| Name | Props | Return | Description |
| ---- | ----- | ------ | ----------- |
|  getStepContent   |   step    |  string      |    identify which step now.          |

###  **Demo**

```jsx

<ExplanationFrame width="100px" height="100px"/>
```
