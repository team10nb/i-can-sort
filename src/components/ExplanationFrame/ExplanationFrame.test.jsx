/*
    Author: Yijie Lu
*/
import { render } from '@testing-library/react';
import ExplanationFrame from './ExplanationFrame';

const mockHandleClick = jest.fn();
    
const data = {
    color:"#4caf50",
    logoFinal:'../../resource/ForLoop.png',
    stepNumber:1,
    picHeight:"350px",
    avatarMarginLeft:"9px",
    avatarMarginTop:"9px",
    handleOpen: mockHandleClick,
  };

test('should contain 1 buttons', () => {
    const {getAllByRole} = render(<ExplanationFrame {...data}/>);
    const buttons = getAllByRole('button');
    expect(buttons.length).toBe(1);
})

test('help button should match Snapshot', () => {
const content = render(<ExplanationFrame {...data} />);
expect(content).toMatchSnapshot();
})
