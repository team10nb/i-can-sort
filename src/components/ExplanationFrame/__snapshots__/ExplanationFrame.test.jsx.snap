// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help button should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="height: 48px;"
      >
        <button
          aria-label="Help"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          title="Help"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          aria-hidden="true"
          class="MuiBackdrop-root makeStyles-backdrop-7"
          style="opacity: 0; visibility: hidden;"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-8 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              style="margin-left: 9px; margin-top: 9px;"
            >
              <button
                aria-label="close"
                class="MuiButtonBase-root MuiIconButton-root"
                index="2"
                tabindex="0"
                type="button"
                value="0"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="color: rgb(76, 175, 80); opacity: 0.8; font-size: 40px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div>
              <div>
                <button
                  class="MuiButtonBase-root MuiCardActionArea-root"
                  style="height: 350px; margin-bottom: 15px;"
                  tabindex="0"
                  type="button"
                >
                  <img
                    alt="logoFinal"
                    src="../../resource/ForLoop.png"
                    width="587px"
                  />
                  <div
                    class="MuiCardContent-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                    />
                  </div>
                  <span
                    class="MuiCardActionArea-focusHighlight"
                  />
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="height: 48px;"
    >
      <button
        aria-label="Help"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        title="Help"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-9"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-backdrop-7"
        style="opacity: 0; visibility: hidden;"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-card-8 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            style="margin-left: 9px; margin-top: 9px;"
          >
            <button
              aria-label="close"
              class="MuiButtonBase-root MuiIconButton-root"
              index="2"
              tabindex="0"
              type="button"
              value="0"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  style="color: rgb(76, 175, 80); opacity: 0.8; font-size: 40px;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div>
            <div>
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                style="height: 350px; margin-bottom: 15px;"
                tabindex="0"
                type="button"
              >
                <img
                  alt="logoFinal"
                  src="../../resource/ForLoop.png"
                  width="587px"
                />
                <div
                  class="MuiCardContent-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                  />
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
